### DISCUSSION TITLE
Why isn't my text appearing inside the rectangle even though they have entirely different color?

### DISCUSSION THREAD

**Discussion by timelessnesses** [slint-ui/slint#6879] [Q&A] [answered]
so here's my slint code
```slint
import "./Sen-VariableFont_wght.ttf";

export component MainWindow inherits Window {
    title: "Contrasity";
    width: 858px;
    height: 520px;

    in property <color> background-color;
    in property <color> text-color;
    in property <float> rating;
    in property <bool> pass-AA;
    in property <bool> pass-AAA;
    in property <bool> pass-AALarge;
    in property <string> font;

    background: black;
    text-color: cyan;
    font: "Sen";
    
   // ...
    
    Rectangle {
        background: text-color;
        border-width: 5px;
        x: 511px;
        y: 186px;
        width: 133px;
        height: 47px;
        border-radius: 5px;
    
        Text {
            font-size: 30px;
            text: "Pass";
            font-italic: false;
            vertical-alignment: center;
            font-family: font;
            color: background-color;
        }
    }
    
}
```
the text is somehow isn't appearing but it is clickable on the Slint live preview  
![image](https://github.com/user-attachments/assets/8bbd4033-2d99-44d9-9692-6314aa59afa9)
I have tried to change my variable name but it didn't help, until I actually not using the `color` variable in text then the text then appears  
![image](https://github.com/user-attachments/assets/da479ec9-a7a7-4e7e-b00f-982e38aa779d)  
When making the font bigger it actually doesn't show up at all?  
![image](https://github.com/user-attachments/assets/dc401371-d57b-482d-8a9e-e22a0d6491d6)

**Comment by hunger** [comment]
I think it is related to declaring properties and only then assigning into them...

If I avoid that it works for me:

```slint
export component MainWindow inherits Window {
    width: 300px;
    height: 200px;

    in property <color> background-color: black;
    in property <color> text-color: cyan;
    
    Rectangle {
        background: text-color;
    
        Text {
            text: "Pass";
            vertical-alignment: center;
            color: background-color;
        }
    } 
}
```

This works because it actually changes `background-color` from its default value (fully transparent) to black.

your example only sets the window background to `black` and leaves the `background-color` unassigned.

**Reply by tronical** [reply, marked as answer]
To clarify further:

 - The provided example uses `color: background-color;` for the `Text` element.
 - The component declares `in property <color> background-color;`
 - However, the initialisation uses a different property: `background: black;`. Therefore `background-color` remains initialised to transparent.
 - Change  `background: black;` to ` `background-color: black;` and the text isn't shown transparent anymore.

**Reply by hunger** [reply]
You might want to keep `background: black;` in addition to `background-color: black;`. The first sets the background color of the window.
